Experimental/CameraComponent/FirstPersonCamera.cs

```diff
@@ -0,0 +1,264 @@
+                }
+                
+                direction.X *= -1; 
+                Yaw = (MathF.PI / 2f) + MathF.Atan2(direction.Z, direction.X);
+                Pitch = MathF.Asin(direction.Y);                    
+            }
+        }
+        public bool UseAnimations
+        {
+            get => useAnimations;
+            set
+            {
+                if(value != useAnimations)
+                {
+                    useAnimations = value;
+                    // turn animations off
+                    if (!value)
+                    {
+                        propertySet.TryGetVector3("Position", out position);
+                        propertySet.TryGetScalar("Yaw", out yaw);
+                        propertySet.TryGetScalar("Pitch", out pitch);
+                        propertySet.TryGetScalar("Roll", out roll);
+                        RaisePropertyChanged("ViewMatrix");
+                    }
+                    // turn animations on
+                    else
+                    {
+                        RaisePropertyChanged("ViewMatrix");
+                    }
+                }
+            }
+        }
+        public Projection Projection
+        {
+            get => projection;
+            set
+            {
+                projection = value;
+                projection.PropertyChanged += PropertyChanged;
+                RaisePropertyChanged("Projection");
+            }
+        }
+        public event PropertyChangedEventHandler PropertyChanged;
+        public void RaisePropertyChanged(string name)
+        {
+            if(PropertyChanged != null)
+            {
+                PropertyChanged(this, new PropertyChangedEventArgs(name));
+            }
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////////////////////
+        /// PUBLIC FUNCTIONS
+        /////////////////////////////////////////////////////////////////////////////////////////////////
+        public Matrix4x4 CreateViewMatrix()
+        {
+            Matrix4x4 transformation = Matrix4x4.Identity;
+            Matrix4x4 matPos = Matrix4x4.CreateTranslation(-position);
+            Matrix4x4 matRoll = Matrix4x4.CreateFromAxisAngle(new Vector3(0, 0, 1), -roll);
+            Matrix4x4 matPitch = Matrix4x4.CreateFromAxisAngle(new Vector3(1, 0, 0), -pitch);
+            Matrix4x4 matYaw = Matrix4x4.CreateFromAxisAngle(new Vector3(0, 1, 0), -yaw);
+
+            return matPos * matYaw * matPitch * matRoll;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////////////////////
+        /// ANIMATION FUNCTIONS
+        ///////////////////////////////////////////////////////////////////////////////////////////////// 
+        public void CreateExpressionAnimation(CompositionPropertySet toAnimate, string propertyName)
+        {
+            propertySet = compositor.CreatePropertySet();

```

Comment:
Should this initialization happen sooner?  If you turn on UseAnimations too early, will it crash?